tests/source.rs

```diff
@@ -391,7 +413,7 @@ fn parse_test(line: &str) -> Option<Result<TestCase>> {
     if let Some(caps) = DIRECTIVE_RE.captures(line) {
         let directive = &caps["directive"];
         let result = parse_directive(directive, &caps["args"], caps.name("negated").is_some())
-            .chain_err(|| ErrorKind::InvalidDirective(line.into()));
+            .map_err(|e| e.context(InvalidDirective { d: line.into() }).into());
```

Comment:
I think the context is backwards here. I think that `parse_test` should return `Option<Result<TestCase, InvalidDirective>>`. Make sure to set the context of the `InvalidDirective` error to be the original error. The tests should then check that `err.cause()` is the expected type.

Reply:
yeah, this is sorta what i'm thinking as well. been toying with this...